casper/src/main/scala/coop/rchain/casper/genesis/Genesis.scala

```diff
@@ -5,75 +5,169 @@ import cats.implicits._
 
 import com.google.protobuf.ByteString
 
-import coop.rchain.catscontrib.Capture
+import coop.rchain.catscontrib._
+import coop.rchain.casper.genesis.contracts.{Rev, Wallet}
 import coop.rchain.casper.protocol._
-import coop.rchain.casper.util.ProtoUtil.{blockHeader, unsignedBlockProto}
-import coop.rchain.casper.util.Sorting
+import coop.rchain.casper.util.ProtoUtil.{blockHeader, termDeploy, unsignedBlockProto}
+import coop.rchain.casper.util.{EventConverter, Sorting}
+import coop.rchain.casper.util.rholang.RuntimeManager
+import coop.rchain.casper.util.rholang.RuntimeManager.StateHash
 import coop.rchain.crypto.codec.Base16
 import coop.rchain.crypto.signatures.Ed25519
-import coop.rchain.shared.{Log, LogSource}
+import coop.rchain.rholang.collection.LinkedList
+import coop.rchain.rholang.interpreter.Runtime
+import coop.rchain.rholang.math.NonNegativeNumber
+import coop.rchain.rholang.mint.MakeMint
+import coop.rchain.rholang.wallet.BasicWallet
+import coop.rchain.shared.{Log, LogSource, Time}
 
 import java.io.{File, PrintWriter}
-import java.nio.file.Path
+import java.nio.file.{Files, Path}
 
+import monix.execution.Scheduler
+
+import scala.concurrent.SyncVar
 import scala.io.Source
 import scala.util.{Failure, Success, Try}
 
 object Genesis {
 
   private implicit val logSource: LogSource = LogSource(this.getClass)
 
-  def fromBondsFile[F[_]: Monad: Capture: Log](maybePath: Option[String],
-                                               numValidators: Int,
-                                               validatorsPath: Path): F[BlockMessage] =
-    for {
-      bondsFile <- toFile[F](maybePath, validatorsPath)
-      bonds     <- getBonds[F](bondsFile, numValidators, validatorsPath)
-    } yield fromBonds(bonds)
+  def withContracts(initial: BlockMessage,
+                    wallets: Seq[Wallet],
+                    startHash: StateHash,
+                    runtimeManager: RuntimeManager)(implicit scheduler: Scheduler): BlockMessage = {
+
+    val blessedTerms = List(
+      LinkedList.term,
+      NonNegativeNumber.term,
+      MakeMint.term,
+      BasicWallet.term,
+      (new Rev(wallets)).term
+    ).map(termDeploy)
+
+    val Right(checkpoint) = runtimeManager.computeState(startHash, blessedTerms)
```

Comment:
Does `computeState` return `Right` always?

```scala
scala> def returnLeft(i: Int): Either[Int, String] = Left(i)
returnLeft: (i: Int)Either[Int,String]

scala> val Right(r) = returnLeft(1)
scala.MatchError: Left(1) (of class scala.util.Left)
  ... 36 elided
```

Reply:
It would be `Left` if injecting a term with the reducer were to fail: https://github.com/rchain/rchain/blob/dev/casper/src/main/scala/coop/rchain/casper/util/rholang/RuntimeManager.scala#L76 . I don't know under what conditions that happens, but I think in that case something is really wrong and having an exception raised probably good. Though I suppose having that error manifest itself in the form of a match error is not very intuitive.